ui/src/app/job-list/job-list.component.ts

```diff
@@ -151,6 +170,26 @@ export class JobListComponent implements OnInit {
     this.jobStream.loadAtLeast((e.pageIndex+2) * e.pageSize)
       .catch((error) => this.handleError(error));
   }
+
+  toggleDisplayColumn(field: DisplayField) {
+    const newValue = !field.primary;
+    this.displayFields.forEach((df) => {
+      if (df.field == field.field) {
+        df.primary = newValue;
+      }
+    })
+  }
+
+  saveSettings() {
+    let fields: string[] = [];
+    this.displayFields.forEach((field) => {
```

Comment:
```
fields = this.displayFields.filter(...)
```

Reply:
How would that work? `fields` is an array of string and `this.displayFields` is an array of DisplayField.

Reply:
Good point, I missed that it was pulling out a subfield. filter + map would work, but I'm not sure I would recommend that over your current approach.